src/core/src/pattern/op/pattern.cpp

```diff
@@ -91,136 +84,314 @@ PatternValueMap as_pattern_value_map(const PatternMap& pattern_map) {
+    size_t pos = 0, pos_next;
+    std::string token;
+    while ((pos_next = s.find(',', pos)) != std::string::npos) {
+        token = s.substr(pos, pos_next - pos);
+        parsed.push_back(token);
+        pos = pos_next + 1;
+    }
+    // collect whole string if no delimiter is found
+    token = s.substr(pos, pos_next);
+    parsed.push_back(token);
+
+    std::vector<std::pair<int64_t, std::string>> idx_to_name;
+
+    bool ellipsis_visited = false;
+    for (int64_t i = 0; i < static_cast<int64_t>(parsed.size()); ++i) {
+        auto dimension = parsed[i];
+        if (dimension.find("...") != std::string::npos) {
+            OPENVINO_ASSERT(!ellipsis_visited, "Only one ellipsis is allowed for symbolic predicate notation");
+            OPENVINO_ASSERT(ends_with(dimension, "..."), "Ellipsis usage is incorrect in symbolic predicate notation");
+            ellipsis_visited = true;
+            idx_to_name.emplace_back(i, dimension);
+            continue;
+        }
+        idx_to_name.emplace_back((ellipsis_visited ? i - static_cast<int64_t>(parsed.size()) : i), dimension);
+    }
+    return {idx_to_name, (ellipsis_visited ? -2 : idx_to_name.size())};
 }
 
-pass::pattern::op::ValuePredicate operator&&(const pass::pattern::op::ValuePredicate& a,
-                                             const pass::pattern::op::ValuePredicate& b) {
-    return [=](const Output<Node>& value) {
-        return a(value) && b(value);
-    };
+std::pair<bool, int64_t> str2int(const std::string& str) {
+    auto s = str.c_str();
+    char* end;
+    int64_t l;
+    l = strtol(s, &end, 10);
+    if (*s == '\0' || *end != '\0')
+        return {1, 0};
+    return {0, l};
 }
 
-pass::pattern::op::ValuePredicate operator&&(const std::function<bool(Output<Node>)>& a,
-                                             const std::function<bool(Output<Node>)>& b) {
-    return [=](const Output<Node>& value) {
-        return a(value) && b(value);
-    };
-}
+struct GroupDetails {
+    std::string name;
+    int64_t begin = 0, end = 0;
 
-pass::pattern::op::ValuePredicate operator&&(const pass::pattern::op::NodePredicate& a,
-                                             const pass::pattern::op::NodePredicate& b) {
-    return pass::pattern::op::as_value_predicate(a) && pass::pattern::op::as_value_predicate(b);
-}
+    size_t size() const {
+        return static_cast<size_t>(end - begin);
+    }
+};
 
-pass::pattern::op::ValuePredicate operator&&(const pass::pattern::op::ValuePredicate& a,
-                                             const pass::pattern::op::NodePredicate& b) {
-    return a && pass::pattern::op::as_value_predicate(b);
-}
+}  // namespace
 
-pass::pattern::op::ValuePredicate operator&&(const pass::pattern::op::NodePredicate& a,
-                                             const pass::pattern::op::ValuePredicate& b) {
-    return pass::pattern::op::as_value_predicate(a) && b;
+op::Predicate shape_matches(const std::string& shape_notation) {
```

Comment:
As it is used for literal value would be better to have variant which accepts rvalue also. 
maybe for literals the string view will be better then creates strings would be not required.
Can input be a partial shape, also?